doc/architecture.md

```diff
@@ -0,0 +1,136 @@
+
+* Content includes Dashboards, Reports, Icons, Symptoms, Alerts, Traversals, others.
+* Metadata includes the describe.xml file that describes an adapter, and the 
+  `manifest.txt` file that describes the management pack as a whole.
+* The Adapter is the code that performs the collection and sends the data to VMware Aria 
+  Operations.
+
+### Adapter Container Image
+
+The adapter container image consists of two parts:
+* A REST server specified by [Collector Framework 2](../vmware_aria_operations_integration_sdk/api/vmware-aria-operations-collector-fwk2.json).
+* The adapter code that performs the collection.
+
+In an adapter created using the SDK, the REST server is provided inside a base container
+image that can built upon. In the adapter's dockerfile, this is specified in the 'FROM' 
+directive on the first line. This default server implements all the required endpoints, 
+provides error handling, and adds some additional context to the user code that is not 
+present in the original REST requests. The server also adds an additional 
+[definition endpoint](../vmware_aria_operations_integration_sdk/api/integration-sdk-definition-endpoint.json)
+which allows the collection code to describe itself, rather than requiring a 
+`describe.xml` file to be manually created, and some features that improve the debugging
+experience.
+
+The server determines how to call the user code by reading a file called `commands.cfg` 
+that is present in the Project Directory. This file provides a mapping from each 
+endpoint served by the REST server to an executable command that will run as a separate 
+process inside the container. Importantly, the server appends two extra arguments to 
+each process call: An input and output named pipe. These are used for interprocess 
+communication.
+
+Because commands are run as a subprocess, the server is able to detect and handle many
+common errors (such as the process crashing without writing a result) without affecting
+the operation of the server.
+
+Importantly, this model enforces that adapters remain stateless. Because each call 
+results in a new subprocess being created, there is no way to save any state between 
+calls in the runtime environment. 
+Some benefits of a stateless adapter are:
+* Potential for (slow) memory leaks is reduced.
+* Potential for an adapter to return stale (cached) data is reduced.
+* Potential for an adapter to crash or hang due to a previous collection leaving the
+  adapter in an inconsistent state is reduced.
+ 
+Note that it is possible to pass data between collections by writing and reading to a 
+file on the container filesystem, but not encouraged. Any restarts of the adapter 
+instance (either by manual action or automatic migration) will create a new container, 
+and any data on the container filesystem will be lost.
+
+![Cloud Proxy Components running two Adapter Container Images](cloud-proxy-components.png)
+
+#### Commands.cfg
```

Comment:
I'm confused about this as a subheader. Although the document mentions it, we haven't previously indicated that `commands.cfg` is an important component, on the same level as the Adapter Container image. This section may be better served as its own page.

Reply:
I think the info is appropriate here, but I've renamed and reorganized the section so it's more clear how it fits in